interpreter/cling/lib/Interpreter/ClingPragmas.cpp

```diff
@@ -68,9 +104,43 @@ namespace {
         = m_Interp.getCI()->getASTContext().getTranslationUnitDecl();
       Sema::ContextAndScopeRAII pushedDCAndS(m_Interp.getSema(),
                                              TU, m_Interp.getSema().TUScope);
-
       Interpreter::PushTransactionRAII pushedT(&m_Interp);
-      m_Interp.loadFile(FileName, true /*allowSharedLib*/);
+      m_Interp.loadFile(file,true /*allowSharedLib*/);
+    }
+  };
+
+  class PHAddIncPath: public PragmaHandler {
+    Interpreter& m_Interp;
+
+  public:
+    PHAddIncPath(Interpreter& interp):
+      PragmaHandler("add_include_path"), m_Interp(interp) {}
+
+    void HandlePragma(Preprocessor &PP,
+                      PragmaIntroducerKind Introducer,
+                      Token &FirstToken) override {
+      // TODO: use Diagnostics!
+      m_Interp.AddIncludePath(HandlePragmaHelper(PP, Introducer, FirstToken, m_Interp,
```

Comment:
If HandlePragmaHelper returns success but an empty string, don't call AddIncludePath() (and don't issue an error message - this might be intentional and it's just fine)